From 6b95d6c373a9f5fa9d43cbaff79a54436de14811 Mon Sep 17 00:00:00 2001
From: Noam Camus <ncamus@habana.ai>
Date: Thu, 23 May 2019 22:15:57 +0300
Subject: [PATCH] [SW-0] sysclk: Gaudi fix PCIe clock consumers

Uart and SPI use PCIe clock (divided by 5)

We use minimal divider for SPI to get maximum performance
Change-Id: I62069d1d6ccba2581e923b4c9846264bf866aa40
---

diff --git a/gaudi/pldm.c b/gaudi/pldm.c
index c0993fb..c07682b 100644
--- a/gaudi/pldm.c
+++ b/gaudi/pldm.c
@@ -11,7 +11,9 @@
 #include <pll.h>
 #include <uart.h>
 
-#define SYS_CLOCK_FREQUENCY	153000ULL
+#define SYS_CLOCK_FREQUENCY	155000ULL
+#define UART_CLOCK_FREQUENCY	(SYS_CLOCK_FREQUENCY / 5)
+#define PLDM_UART_BAUDRATE	1200
 
 bool hl_pll_is_configurable(struct hl_pll_cfg *cfg)
 {
@@ -35,14 +37,12 @@
 
 void hl_uart_init(void)
 {
-	preboot_uart_init(SYS_CLOCK_FREQUENCY / HL_UART_BAUDRATE);
+	preboot_uart_init(UART_CLOCK_FREQUENCY / PLDM_UART_BAUDRATE);
 }
 
 u32 hl_spi_baudr_get(void)
 {
-	u32 factor;
-
-	factor = SYS_CLOCK_FREQUENCY / HL_SPI_FLASH_MAX_FREQ;
+	u32 factor = 2; /* Min Divider */
 
 	/* baud rate doesn't support odd number */
 	return (factor + 1) & 0xFFFE;
